src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,115 @@
-                                          int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
-    }
-
-    /**
-     * Given a List of `Person` instances (having `name`, `age`, `sex` and `cats` fields,
-     * and each `Cat` having a `name` and `age`),
-     * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
-     */
-    public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
-    }
-
-    /**
-     * Your help with a election is needed. Given list of candidates, where each element
-     * has Candidate.class type.
-     * Check which candidates are eligible to apply for president position and return their
-     * names sorted alphabetically.
-     * The requirements are: person should be older than 35 years, should be allowed to vote,
-     * have nationality - 'Ukrainian'
-     * and live in Ukraine for 10 years. For the last requirement use field periodsInUkr,
-     * which has following view: "2002-2015"
-     * We want to reuse our validation in future, so let's write our own impl of Predicate
-     * parametrized with Candidate in CandidateValidator.
-     */
-    public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
-    }
-}
+package practice;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.function.Predicate;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import model.Candidate;
+import model.Cat;
+import model.Person;
+
+public class StreamPractice {
+    /**
+     * Given list of strings where each element contains 1+ numbers:
+     * input = {"5,30,100", "0,22,7", ...}
+     * return min integer value. One more thing - we're interested in even numbers.
+     * If there is no needed data throw RuntimeException with message
+     * "Can't get min value from list: < Here is our input 'numbers' >"
+     */
+    public int findMinEvenNumber(List<String> numbers) {
+        return numbers.stream()
+                .map(number -> number.split(","))
+                .flatMap(Stream::of)
+                .mapToInt(Integer::parseInt)
+                .filter(num -> num % 2 == 0)
+                .min()
+                .orElseThrow(() -> new RuntimeException("Can't get min value from list: "
+                        + numbers));
+    }
+
+    /**
+     * Given a List of Integer numbers,
+     * return the average of all odd numbers from the list or throw NoSuchElementException.
+     * But before that subtract 1 from each element on an odd position (having the odd index).
+     */
+    public Double getOddNumsAverage(List<Integer> numbers) {
+        return IntStream.range(0, numbers.size())
+                        .map(index -> index % 2 != 0 ? numbers.get(index) - 1 : numbers.get(index))

```

Comment:
minor 

it is always better for readability not to use negation when possible
```suggestion
                        .map(index -> index % 2 == 0 ? numbers.get(index): numbers.get(index) - 1 )
```